### ISSUE TITLE
zf1s/zend-ldap doesn't work with PHP8

### ISSUE THREAD

**Issue by marcelopm** [zf1s/zf1#124]
PHP version: 8.0.14
Reproducible by:
- Just make a simple search and iterate though the resultset:
```php
//  code for context purposes only 
$options = [ ... ];
$filter = \Zend_Ldap_Filter::equals('objectclass', 'person');
$base = 'OU=test';
$ldap = new \Zend_Ldap();
$ldap->setOptions($options);
$ldap->bind();
$result = $ldap->search($filter, $base, \Zend_Ldap::SEARCH_SCOPE_SUB);
foreach($result as $item) {
...
```

The above will fail in the foreach with ``` ldap_first_attribute() expects exactly 2 arguments, 3 given``` error due to https://github.com/zf1s/zend-ldap/blob/6eac2a55a31fe0d1994275c504a0f3de1a915280/library/Zend/Ldap/Collection/Iterator/Default.php#L203

Also, zf1s/zend-ldap uses @ operator in a few places, which won't suppress fatal errors anymore in PHP8: https://www.php.net/releases/8.0/en.php#:~:text=the%20%40%20operator%20no%20longer%20silences%20fatal%20errors.

**Comment by marcelopm** [comment]
The following lines are likely to yield an ArgumentCountError exception too:
- https://github.com/zf1s/zend-ldap/blob/master/library/Zend/Ldap/Collection/Iterator/Default.php#L224
-https://github.com/zf1s/zend-ldap/blob/master/library/Zend/Ldap.php#L1560

**Comment by glensc** [comment]
@marcelopm 


🚧 Never link to branches, share a permalink 🚧
- see https://news.ycombinator.com/item?id=8046710
  and https://help.github.com/articles/getting-permanent-links-to-files/
  and https://docs.gitlab.com/ee/user/shortcuts.html#project-files
- hit `y`
- and perhaps for https://github.com/josephfrazier/octopermalinker (GitHub)

**Comment by marcelopm** [comment]
Closing it as ZF1's EOF in 2016

**Comment by glensc** [comment]
1. zf1 is eol
2. this is zf1s project, fork of zf1

https://github.com/zf1s/zf1#readme:

> This is a monorepo of a fork of Zend Framework 1, made after it's reached its EOL

**Comment by falkenhawk** [comment]
> The above will fail in the foreach with ` ldap_first_attribute() expects exactly 2 arguments, 3 given` error due to https://github.com/zf1s/zend-ldap/blob/6eac2a55a31fe0d1994275c504a0f3de1a915280/library/Zend/Ldap/Collection/Iterator/Default.php#L203

@marcelopm 
sigh, according to this https://www.php.net/manual/en/function.ldap-first-attribute.php php8+ should not throw an error
> For backwards compatibility PHP will not throw an error if this parameter is passed. 
but it is not true, apparently.

In order to fix it, we would need to add an ldap server (e.g. openldap) to github workflow https://github.com/zf1s/zf1/blob/c02174bb4c7e04aa6a3a25feb63f3bd65b23e302/.github/workflows/tests.yml and then enable `TESTS_ZEND_LDAP_ONLINE_ENABLED` in https://github.com/zf1s/zf1/blob/c02174bb4c7e04aa6a3a25feb63f3bd65b23e302/tests/TestConfiguration.ci.php and configure it accordingly in that file, based on:
https://github.com/zf1s/zf1/blob/c02174bb4c7e04aa6a3a25feb63f3bd65b23e302/tests/TestConfiguration.dist.php#L368-L417

Unfortunately, we don't have any resources available to do this. Contributions would be very welcome...

**Comment by falkenhawk** [comment]
Fixed in [1.15.2](https://github.com/zf1s/zf1/releases/tag/1.15.2).

**Comment by nikolaipulkkinen** [comment]
hello @marcelopm , faced same issue but https://github.com/rectorphp/rector solved it for me 
![image](https://user-images.githubusercontent.com/1893595/220857907-29134d15-b6c0-4048-83e2-8221ce46ff6e.png)

**Comment by glensc** [comment]
@nikolaipulkkinen this [was said to be fixed in 1.15.2](https://github.com/zf1s/zf1/issues/124#issuecomment-1436124197), if this is new issue, create a new bug report.
